src/Cache/CacheEngine.php

```diff
@@ -107,12 +107,12 @@ protected function ensureValidKey($key): void
     /**
      * Ensure the validity of the argument type and cache keys.
      *
-     * @param iterable $iterable The iterable to check.
+     * @param mixed $iterable The iterable to check.
      * @param string $check Whether to check keys or values.
      * @return void
      * @throws \Cake\Cache\InvalidArgumentException
      */
-    protected function ensureValidType($iterable, string $check = self::CHECK_VALUE): void
+    protected function ensureValidType(mixed $iterable, string $check = self::CHECK_VALUE): void
```

Comment:
This needs to stay as `iterable` as it's called from `*Multiple()` methods which already have `iterable` type.